src/gamekit.cs

```diff
@@ -2163,4 +2186,142 @@ interface GKTurnBasedEventListener
 	}
+
+	[iOS (10,0)][Mac (10,12)]
+	[BaseType (typeof(NSObject))]
+	interface GKGameSession
+	{
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("title")]
+		string Title { get; }
+
+		[Export ("owner")]
+		GKCloudPlayer Owner { get; }
+
+		[Export ("players")]
+		GKCloudPlayer[] Players { get; }
+
+		[Export ("lastModifiedDate")]
+		NSDate LastModifiedDate { get; }
+
+		[Export ("lastModifiedPlayer")]
+		GKCloudPlayer LastModifiedPlayer { get; }
+
+		[Export ("maxNumberOfConnectedPlayers")]
+		nint MaxNumberOfConnectedPlayers { get; }
+
+		[Export ("badgedPlayers")]
+		GKCloudPlayer[] BadgedPlayers { get; }
+
+		[Static]
+		[Export ("createSessionInContainer:withTitle:maxConnectedPlayers:completionHandler:")]
+		void CreateSession (string containerName, string title, nint maxPlayers, Action<GKGameSession, NSError> completionHandler);
+
+		[Static]
+		[Export ("loadSessionsInContainer:completionHandler:")]
+		void LoadSessions (string containerName, Action<GKGameSession[], NSError> completionHandler);
+
+		[Static]
+		[Export ("loadSessionWithIdentifier:completionHandler:")]
+		void LoadSession (string identifier, Action<GKGameSession, NSError> completionHandler);
+
+		[Static]
+		[Export ("removeSessionWithIdentifier:completionHandler:")]
+		void RemoveSession (string identifier, Action<NSError> completionHandler);
+
+		[Export ("getShareURLWithCompletionHandler:")]
+		void GetShareUrl (Action<NSUrl, NSError> completionHandler);
+
+		[Export ("loadDataWithCompletionHandler:")]
+		void LoadData (Action<NSData, NSError> completionHandler);
+
+		[Export ("saveData:completionHandler:")]
+		void SaveData (NSData data, Action<NSData, NSError> completionHandler);
+
+		[Export ("setConnectionState:completionHandler:")]
+		void SetConnectionState (GKConnectionState state, Action<NSError> completionHandler);
+
+		[Export ("playersWithConnectionState:")]
+		GKCloudPlayer[] GetPlayers (GKConnectionState state);
+
+		[Export ("sendData:withTransportType:completionHandler:")]
+		void SendData (NSData data, GKTransportType transport, Action<NSError> completionHandler);
+
+		[Export ("sendMessageWithLocalizedFormatKey:arguments:data:toPlayers:badgePlayers:completionHandler:")]
+		void SendMessage (string key, string[] arguments, NSData data, GKCloudPlayer[] players, bool badgePlayers, Action<NSError> completionHandler);
+
+		[Export ("clearBadgeForPlayers:completionHandler:")]
+		void ClearBadge (GKCloudPlayer[] players, Action<NSError> completionHandler);
+
+		[Static]
+		[Export ("acceptShareURL:handler:")]
+		void AcceptShareUrl (NSUrl url, Action<NSError> handler);
+
+		[Static]
+		[Export ("addEventListener:")]
+		void AddEventListener (IGKGameSessionEventListener listener);
+
+		[Static]
+		[Export ("removeEventListener:")]
+		void RemoveEventListener (IGKGameSessionEventListener listener);
+	}
+
+	interface IGKGameSessionEventListener {}
+
+	[iOS (10,0)][Mac (10,12)]
+	[Protocol, Model]
```

Comment:
drop `[Model]` (and `[BaseType]`) as it's not clear a default implementation is helpful as everything looks optional